sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -135,7 +136,13 @@ public IntervalMetricReader build() {
   @SuppressWarnings("FutureReturnValueIgnored")
   private IntervalMetricReader(InternalState internalState) {
     this.exporter = new Exporter(internalState);
-    this.scheduler = Executors.newScheduledThreadPool(1, MoreExecutors.platformThreadFactory());
+    this.scheduler =
+        Executors.newScheduledThreadPool(
+            1,
+            new ThreadFactoryBuilder()
+                .setDaemon(true)
```

Comment:
since there is a (admittedly rare) security manager-induced failure mode here, can we add a comment to the javadoc for this class about it?

Reply:
Note that if we do this here, then newThread will throw if setDaemon fails.

Reply:
do we know how fast this fails? I think the javadoc addition should be good to document this fact. Might be good to have a custom threadfactory which will allow non-daemon access if it's forbidden. @johnbley any idea how difficult that would be?

Reply:
Probably a better solution:
Implement our own ThreadFactory see what we did in OpenCensus https://github.com/census-instrumentation/opencensus-java/blob/master/impl_core/src/main/java/io/opencensus/implcore/internal/DaemonThreadFactory.java 

And always use that to produce threads :)

Reply:
Definitely strong agree on this one!